test/external/dataplane/gn_connectivity.go

```diff
@@ -0,0 +1,439 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dataplane
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	resourceUtil "github.com/submariner-io/admiral/pkg/resource"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	"github.com/submariner-io/shipyard/test/e2e/tcp"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/globalnet/constants"
+	subFramework "github.com/submariner-io/submariner/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type globalnetTestParams struct {
+	ClusterEgressIPType subFramework.GlobalEgressIPType
+}
+
+/*
+   - Test environment for external network w/ globalnet is as follows:
+
+                  [ext-app]  [gateway-cluster]  [non-gateway-cluster]
+                     |            |       |        |
+    pseudo-ext *-----+------------+--*    |        |
+                                          |        |
+    kind                            *-----+--------+-------*
+
+
+   - For globalnet environment, expected behaviors of connectivity and source IPs are:
+
+       from / to       |    ext-app    |  gateway-cluster  |  non-gateway-cluster  |
+   ------------------- | ------------- | ----------------- | --------------------- |
+   ext-app             |    N/A        |       R(*2)(*4)   |        S(*3)          |
```

Comment:
> When you say S(*3) -> Pod w/ hostnetwork is not reachable when pod isn't on a gateway node -> Does it mean the ext-app is trying to connect to a Pod w/ hostnetwork?

Yes. 

> If it is so, then such use-case may not work even between regular clusters connected via submariner because the hostnetwork pod will have an IPAddress from the underlay network which is not advertised as local subnet to the remote clusters.

I'm not really sure how it works now, but it seems that it works only when the pod is on gateway node. I agree that we don't need to support hostnetwork (either on gateway node or non-gateway node).